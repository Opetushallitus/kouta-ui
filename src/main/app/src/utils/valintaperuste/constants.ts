import { KOULUTUSTYYPPI } from '#/src/constants';

export const koulutustyypitWithValintatapa = [
  KOULUTUSTYYPPI.YLIOPISTOKOULUTUS,
  KOULUTUSTYYPPI.AMKKOULUTUS,
  KOULUTUSTYYPPI.AMMATILLINEN_OPETTAJAKOULUTUS,
  KOULUTUSTYYPPI.AMMATILLINEN_OPINTO_OHJAAJA_KOULUTUS,
  KOULUTUSTYYPPI.AMMATILLINEN_ERITYISOPETTAJA_KOULUTUS,
];
