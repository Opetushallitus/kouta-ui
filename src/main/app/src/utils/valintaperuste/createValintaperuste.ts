import { makeEntityMutator } from '#/src/utils/makeEntityMutator';

export const createValintaperuste = makeEntityMutator(
  'put',
  'valintaperuste',
  'kouta-backend.valintaperuste'
);
