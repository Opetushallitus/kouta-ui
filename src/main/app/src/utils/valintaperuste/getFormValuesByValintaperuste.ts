import _ from 'lodash';

import { parseEditorState } from '#/src/components/Editor/utils';
import { getKokeetTaiLisanaytotValues } from '#/src/utils/form/getKokeetTaiLisanaytotValues';

const parseSisalto = sisalto => {
  if (!_.isArray(sisalto)) {
    return [];
  }

  return sisalto.map(({ tyyppi, data }) => {
    if (tyyppi === 'teksti') {
      return {
        tyyppi,
        data: _.isObject(data) ? _.mapValues(data, parseEditorState) : {},
      };
    }

    return { tyyppi, data };
  });
};

export const getFormValuesByValintaperuste = valintaperuste => {
  const {
    hakutapaKoodiUri = null,
    kielivalinta = [],
    kohdejoukkoKoodiUri = null,
    nimi = {},
    metadata = {},
    koulutustyyppi = null,
    onkoJulkinen = false,
    tila,
    sorakuvausId,
    valintakokeet,
  } = valintaperuste;

  const {
    valintatavat = [],
    kuvaus = {},
    sisalto = [],
    valintakokeidenYleiskuvaus,
  } = metadata;

  return {
    tila,
    perustiedot: {
      tyyppi: koulutustyyppi,
      kieliversiot: kielivalinta,
      hakutapa: hakutapaKoodiUri,
      kohdejoukko: kohdejoukkoKoodiUri ? { value: kohdejoukkoKoodiUri } : null,
    },
    julkinen: onkoJulkinen,
    kuvaus: {
      nimi,
      kuvaus: _.mapValues(kuvaus || {}, parseEditorState),
      sisalto: parseSisalto(sisalto),
    },
    valintatavat: (valintatavat || []).map(
      ({
        nimi: valintatapaNimi,
        kuvaus: valintatapaKuvaus,
        sisalto: valintatapaSisalto,
        valintatapaKoodiUri,
        kynnysehto,
        enimmaispisteet,
        vahimmaispisteet,
      }) => ({
        nimi: valintatapaNimi || {},
        kuvaus: valintatapaKuvaus || {},
        sisalto: parseSisalto(valintatapaSisalto),
        tapa: valintatapaKoodiUri ? { value: valintatapaKoodiUri } : null,
        kynnysehto: _.mapValues(kynnysehto || {}, parseEditorState),
        enimmaispistemaara:
          _.toString(enimmaispisteet)?.replace('.', ',') || '',
        vahimmaispistemaara:
          _.toString(vahimmaispisteet)?.replace('.', ',') || '',
      })
    ),
    soraKuvaus: sorakuvausId
      ? {
          value: sorakuvausId,
        }
      : null,
    valintakokeet: getKokeetTaiLisanaytotValues(
      valintakokeet,
      valintakokeidenYleiskuvaus
    ),
  };
};
