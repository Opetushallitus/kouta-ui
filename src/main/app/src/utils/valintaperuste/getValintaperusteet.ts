export const getValintaperusteet = async ({
  organisaatioOid,
  hakuOid,
  koulutustyyppi,
  httpClient,
  apiUrls,
}) => {
  const { data } = await httpClient.get(
    apiUrls.url('kouta-backend.valintaperuste-list'),
    {
      params: {
        organisaatioOid,
        ...(hakuOid && { hakuOid }),
        ...(koulutustyyppi && { koulutustyyppi }),
        myosArkistoidut: false,
      },
    }
  );

  return data;
};
