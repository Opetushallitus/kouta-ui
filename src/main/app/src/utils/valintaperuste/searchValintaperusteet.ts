export const searchValintaperusteet = async ({
  httpClient,
  apiUrls,
  params,
}) => {
  const { data } = await httpClient.get(
    apiUrls.url('kouta-backend.search.valintaperusteet'),
    {
      params,
      errorNotifier: {
        silent: true,
      },
    }
  );

  return data;
};
