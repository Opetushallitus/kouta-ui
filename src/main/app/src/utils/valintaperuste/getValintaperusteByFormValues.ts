import produce from 'immer';
import _ from 'lodash';

import { serializeEditorState } from '#/src/components/Editor/utils';
import { isNumeric, isDeepEmptyFormValues, parseFloatComma } from '#/src/utils';
import { getKokeetTaiLisanaytotData } from '#/src/utils/form/getKokeetTaiLisanaytotData';

import { KOULUTUSTYYPIT_WITH_VALINTATAPA } from './constants';

const getArrayValue = (values, key) => {
  const valueCandidate = _.get(values, key);
  return isDeepEmptyFormValues(valueCandidate) ? [] : valueCandidate;
};

const serializeTable = ({ table, kielivalinta }) => {
  if (!table?.rows) {
    return { rows: [] };
  }

  return produce(table, draft => {
    (draft.rows || []).forEach((row, rowIndex) => {
      if (_.isObject(row)) {
        row.index = rowIndex;

        (row.columns || []).forEach((column, columnIndex) => {
          if (_.isObject(column)) {
            column.index = columnIndex;

            if (_.isObject(column.text)) {
              column.text = _.pick(column.text, kielivalinta);
            }
          }
        });
      }
    });
  });
};

const serializeSisalto = (sisalto, kielivalinta = []) => {
  if (!_.isArray(sisalto)) {
    return [];
  }

  return sisalto.map(({ tyyppi, data }) => {
    let serializedData = {};

    if (tyyppi === 'teksti') {
      serializedData = _.pick(
        _.isObject(data) ? _.mapValues(data, serializeEditorState) : {},
        kielivalinta
      );
    }

    if (tyyppi === 'taulukko') {
      serializedData = serializeTable({ table: data, kielivalinta });
    }

    return {
      tyyppi,
      data: serializedData,
    };
  });
};

export const getValintaperusteByFormValues = values => {
  const { tila, muokkaaja, perustiedot, esikatselu = false } = values;

  const koulutustyyppi = perustiedot?.tyyppi ?? null;
  const julkinen = Boolean(values?.julkinen);

  const hakutapaKoodiUri = perustiedot?.hakutapa;
  const kielivalinta = perustiedot?.kieliversiot ?? [];
  const kohdejoukkoKoodiUri = perustiedot?.kohdejoukko?.value ?? null;
  const nimi = _.pick(values?.kuvaus?.nimi, kielivalinta);

  const kuvaus = _.pick(
    _.mapValues(values?.kuvaus?.kuvaus ?? {}, serializeEditorState),
    kielivalinta
  );
  const hakukelpoisuus = _.mapValues(
    _.pick(values?.hakukelpoisuus || {}, kielivalinta),
    serializeEditorState
  );
  const lisatiedot = _.mapValues(
    _.pick(values?.lisatiedot || {}, kielivalinta),
    serializeEditorState
  );
  const sisalto = serializeSisalto(values?.kuvaus?.sisalto, kielivalinta);

  const valintatavat = KOULUTUSTYYPIT_WITH_VALINTATAPA.includes(koulutustyyppi)
    ? getArrayValue(values, 'valintatavat').map(
        ({
          nimi: valintatapaNimi,
          kuvaus: valintatapaKuvaus,
          sisalto: valintatapaSisalto,
          tapa,
          kynnysehto,
          enimmaispistemaara,
          vahimmaispistemaara,
        }) => ({
          nimi: _.pick(valintatapaNimi || {}, kielivalinta),
          kuvaus: _.pick(valintatapaKuvaus || {}, kielivalinta),
          sisalto: serializeSisalto(valintatapaSisalto, kielivalinta),
          valintatapaKoodiUri: tapa?.value,
          kaytaMuuntotaulukkoa: false,
          kynnysehto: _.mapValues(
            _.pick(kynnysehto || {}, kielivalinta),
            serializeEditorState
          ),
          enimmaispisteet: isNumeric(enimmaispistemaara)
            ? parseFloatComma(enimmaispistemaara)
            : null,
          vahimmaispisteet: isNumeric(vahimmaispistemaara)
            ? parseFloatComma(vahimmaispistemaara)
            : null,
        })
      )
    : [];

  const valintakokeidenYleiskuvaus = _.mapValues(
    values?.valintakokeet?.yleisKuvaus,
    serializeEditorState
  );

  const valintakokeet = getKokeetTaiLisanaytotData({
    valintakoeValues: values?.valintakokeet,
    kielivalinta,
  });

  return {
    organisaatioOid: values?.organisaatioOid?.value,
    externalId: _.isEmpty(values?.externalId) ? null : values?.externalId,
    tila,
    muokkaaja,
    kielivalinta,
    hakutapaKoodiUri,
    kohdejoukkoKoodiUri,
    nimi,
    koulutustyyppi,
    julkinen,
    valintakokeet,
    metadata: {
      tyyppi: koulutustyyppi,
      valintatavat,
      kuvaus,
      sisalto,
      valintakokeidenYleiskuvaus,
      hakukelpoisuus,
      lisatiedot,
    },
    esikatselu,
  };
};
