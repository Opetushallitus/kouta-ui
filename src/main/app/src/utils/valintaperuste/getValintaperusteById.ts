import { ENTITY } from '#/src/constants';

import { getEntityByOid, useEntityByOid } from '../api/getEntityByOid';

export const getValintaperusteById = async ({ oid, apiUrls, httpClient }) =>
  getEntityByOid({
    entityType: ENTITY.VALINTAPERUSTE,
    oid,
    apiUrls,
    httpClient,
  });

export const useValintaperusteById = (id?: string | null, options = {}) =>
  useEntityByOid(ENTITY.VALINTAPERUSTE, id, options);

export default getValintaperusteById;
