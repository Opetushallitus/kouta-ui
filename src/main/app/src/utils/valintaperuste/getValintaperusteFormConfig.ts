import { KOULUTUSTYYPIT, KOULUTUSTYYPPI } from '#/src/constants';
import createFormConfigBuilder from '#/src/utils/form/createFormConfigBuilder';
import {
  validateIfJulkaistu,
  getKielivalinta,
  pohjaValintaSectionConfig,
  julkinenSectionConfig,
  valintakokeetSection,
  validateOptionalTranslatedField,
  validateRelations,
} from '#/src/utils/form/formConfigUtils';

const koulutustyypitWithValintatapa = [
  KOULUTUSTYYPPI.YLIOPISTOKOULUTUS,
  KOULUTUSTYYPPI.AMKKOULUTUS,
  KOULUTUSTYYPPI.AMMATILLINEN_OPETTAJAKOULUTUS,
  KOULUTUSTYYPPI.AMMATILLINEN_OPINTO_OHJAAJA_KOULUTUS,
  KOULUTUSTYYPPI.AMMATILLINEN_ERITYISOPETTAJA_KOULUTUS,
];

const config = createFormConfigBuilder().registerSections([
  pohjaValintaSectionConfig,
  {
    section: 'perustiedot',
    koulutustyypit: KOULUTUSTYYPIT,
    parts: [
      {
        field: '.tyyppi',
        validate: eb => eb.validateExistence('perustiedot.tyyppi'),
        required: true,
      },
      {
        field: '.kieliversiot',
        validate: eb =>
          eb.validateArrayMinLength('perustiedot.kieliversiot', 1),
        required: true,
      },
      {
        field: '.hakutapa',
        validate: validateIfJulkaistu(eb =>
          eb.validateExistence('perustiedot.hakutapa')
        ),
        required: true,
      },
      {
        field: '.kohdejoukko',
        validate: validateIfJulkaistu(eb =>
          eb.validateExistence('perustiedot.kohdejoukko')
        ),
        required: true,
      },
    ],
  },
  {
    section: 'kuvaus',
    koulutustyypit: KOULUTUSTYYPIT,
    fields: {
      '.nimi': {
        validate: (eb, values) =>
          eb.validateTranslations('kuvaus.nimi', getKielivalinta(values)),
        required: true,
      },
      '.kuvaus': {
        validate: validateOptionalTranslatedField('kuvaus.kuvaus'),
      },
    },
  },
  {
    section: 'valintatavat',
    koulutustyypit: koulutustyypitWithValintatapa,
    fields: {
      valintatavat: {
        validate: validateIfJulkaistu(eb =>
          eb
            .validateArrayMinLength('valintatavat', 1, {
              isFieldArray: true,
            })
            .validateArray('valintatavat', eb =>
              eb
                .validateExistence('tapa')
                .validateTranslations('nimi', getKielivalinta(eb.getValues()))
            )
        ),
      },
      '.tapa': {
        required: true,
      },
      '.nimi': {
        required: true,
      },
    },
  },
  {
    section: 'soraKuvaus',
    koulutustyypit: KOULUTUSTYYPIT,
    field: 'soraKuvaus',
  },
  julkinenSectionConfig,
  {
    section: 'tila',
    koulutustyypit: KOULUTUSTYYPIT,
    field: 'tila',
    required: true,
    validate: (eb, values) =>
      validateRelations([
        {
          key: 'soraKuvaus',
          t: 'yleiset.soraKuvaus',
        },
      ])(eb.validateExistence('tila'), values),
  },
  valintakokeetSection,
  {
    section: 'esikatselu',
    field: 'esikatselu',
    koulutustyypit: KOULUTUSTYYPIT,
  },
]);

const getValintaperusteFormConfig = koulutustyyppi => {
  return config.getConfig(koulutustyyppi);
};

export default getValintaperusteFormConfig;
