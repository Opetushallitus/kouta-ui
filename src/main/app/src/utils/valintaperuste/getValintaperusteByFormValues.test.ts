import { parseEditorState } from '#/src/components/LexicalEditorUI/utils';
import { getValintaperusteByFormValues } from '#/src/utils/valintaperuste/getValintaperusteByFormValues';

import {
  BASE_VALINTAPERUSTE_FORM_DATA,
  valintakokeetWithExtraTranslations,
  valintatapa,
  valintatapaWithExtraTranslations,
} from '../testFormData';

test('Should convert valintaperuste form with valintatapa', () => {
  const data = {
    ...BASE_VALINTAPERUSTE_FORM_DATA,
    valintatavat: [valintatapa],
  };

  const valintaperuste = getValintaperusteByFormValues(data);

  expect(valintaperuste).toMatchSnapshot();
});

test('Should not convert translations for non-selected language', () => {
  const data = {
    ...BASE_VALINTAPERUSTE_FORM_DATA,
    valintatavat: [valintatapaWithExtraTranslations],
    valintakokeet: valintakokeetWithExtraTranslations,
  };

  const valintaperuste = getValintaperusteByFormValues(data);

  expect(valintaperuste).toMatchSnapshot();
});

test('Should convert valintaperuste form without valintatapa', () => {
  const valintaperuste = getValintaperusteByFormValues({
    ...BASE_VALINTAPERUSTE_FORM_DATA,
    valintatavat: [{}],
  });

  expect(valintaperuste).toMatchSnapshot();
});
