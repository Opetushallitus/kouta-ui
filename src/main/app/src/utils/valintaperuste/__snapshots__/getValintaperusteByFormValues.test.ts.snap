// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should convert valintaperuste form with valintatapa 1`] = `
Object {
  "esikatselu": false,
  "hakutapaKoodiUri": "tapa_1#1",
  "julkinen": true,
  "kielivalinta": Array [
    "fi",
    "sv",
  ],
  "kohdejoukkoKoodiUri": "joukko_1#1",
  "koulutustyyppi": "amk",
  "metadata": Object {
    "hakukelpoisuus": Object {
      "fi": "<h1>Fi hakukelpoisuus</h1>",
      "sv": "<h1>Sv hakukelpoisuus</h1>",
    },
    "kuvaus": Object {
      "fi": "<h1>Fi kuvaus</h1>",
      "sv": "<h1>Sv kuvaus</h1>",
    },
    "lisatiedot": Object {
      "fi": "<h1>Fi lisatiedot</h1>",
      "sv": "<h1>Sv lisatiedot</h1>",
    },
    "sisalto": Array [],
    "tyyppi": "amk",
    "valintakokeidenYleiskuvaus": Object {
      "fi": "<p>Yleiskuvaus - fi</p>",
      "sv": "<p>Yleiskuvaus - sv</p>",
    },
    "valintatavat": Array [
      Object {
        "enimmaispisteet": 20.2,
        "kaytaMuuntotaulukkoa": false,
        "kuvaus": Object {
          "fi": "Fi kuvaus",
          "sv": "Sv kuvaus",
        },
        "kynnysehto": Object {
          "fi": "<p>Fi kynnysehto</p>",
          "sv": "<p>Sv kynnysehto</p>",
        },
        "nimi": Object {
          "fi": "Fi nimi",
          "sv": "Sv nimi",
        },
        "sisalto": Array [
          Object {
            "data": Object {
              "fi": "<h2>Fi sisalto</h2>",
              "sv": "<h2>Sv sisalto</h2>",
            },
            "tyyppi": "teksti",
          },
          Object {
            "data": Object {
              "rows": Array [
                Object {
                  "columns": Array [
                    Object {
                      "index": 0,
                      "text": Object {
                        "fi": "Fi column1",
                        "sv": "Sv column1",
                      },
                    },
                    Object {
                      "index": 1,
                      "text": Object {
                        "fi": "Fi column2",
                        "sv": "Sv column2",
                      },
                    },
                  ],
                  "index": 0,
                },
                Object {
                  "columns": Array [
                    Object {
                      "index": 0,
                      "text": Object {
                        "fi": "Fi column3",
                        "sv": "Sv column3",
                      },
                    },
                    Object {
                      "index": 1,
                      "text": Object {
                        "fi": "Fi column4",
                        "sv": "Sv column4",
                      },
                    },
                  ],
                  "index": 1,
                },
              ],
            },
            "tyyppi": "taulukko",
          },
        ],
        "vahimmaispisteet": 10.1,
        "valintatapaKoodiUri": "tapa_1#1",
      },
    ],
  },
  "muokkaaja": undefined,
  "nimi": Object {
    "fi": "Fi nimi",
    "sv": "Sv nimi",
  },
  "tila": "julkaistu",
  "valintakokeet": Array [
    Object {
      "id": undefined,
      "metadata": Object {
        "erityisjarjestelytMahdollisia": true,
        "liittyyEnnakkovalmistautumista": true,
        "ohjeetEnnakkovalmistautumiseen": Object {
          "fi": "<p>Ohjeet ennakkovalmistautumiseen - fi</p>",
          "sv": "<p>ohjeet ennakkovalmistautumiseen - sv</p>",
        },
        "ohjeetErityisjarjestelyihin": Object {
          "fi": "<p>Ohjeet erityisj채rjestelyihin - fi</p>",
          "sv": "<p>Ohjeet erityisj채rjestelyihin - sv</p>",
        },
        "tietoja": Object {
          "fi": "<p>Tietoa hakijalle - fi</p>",
          "sv": "<p>Tietoa hakijalle - sv</p>",
        },
        "vahimmaispisteet": 30.4,
      },
      "nimi": Object {
        "fi": "nimi - fi",
        "sv": "nimi - sv",
      },
      "tilaisuudet": Array [
        Object {
          "aika": Object {
            "alkaa": "2019-04-16T08:44",
            "paattyy": "2019-04-18T08:44",
          },
          "jarjestamispaikka": Object {
            "fi": "jarjestamispaikka - fi",
            "sv": "jarjestamispaikka - sv",
          },
          "lisatietoja": Object {
            "fi": "<p>fi lisatietoja</p>",
            "sv": "<p>sv lisatietoja</p>",
          },
          "osoite": Object {
            "osoite": Object {
              "fi": "fi osoite",
              "sv": "sv osoite",
            },
            "postinumeroKoodiUri": "posti_1#1",
          },
        },
      ],
      "tyyppiKoodiUri": "tyyppi_1#1",
    },
  ],
}
`;

exports[`Should convert valintaperuste form without valintatapa 1`] = `
Object {
  "esikatselu": false,
  "hakutapaKoodiUri": "tapa_1#1",
  "julkinen": true,
  "kielivalinta": Array [
    "fi",
    "sv",
  ],
  "kohdejoukkoKoodiUri": "joukko_1#1",
  "koulutustyyppi": "amk",
  "metadata": Object {
    "hakukelpoisuus": Object {
      "fi": "<h1>Fi hakukelpoisuus</h1>",
      "sv": "<h1>Sv hakukelpoisuus</h1>",
    },
    "kuvaus": Object {
      "fi": "<h1>Fi kuvaus</h1>",
      "sv": "<h1>Sv kuvaus</h1>",
    },
    "lisatiedot": Object {
      "fi": "<h1>Fi lisatiedot</h1>",
      "sv": "<h1>Sv lisatiedot</h1>",
    },
    "sisalto": Array [],
    "tyyppi": "amk",
    "valintakokeidenYleiskuvaus": Object {
      "fi": "<p>Yleiskuvaus - fi</p>",
      "sv": "<p>Yleiskuvaus - sv</p>",
    },
    "valintatavat": Array [],
  },
  "muokkaaja": undefined,
  "nimi": Object {
    "fi": "Fi nimi",
    "sv": "Sv nimi",
  },
  "tila": "julkaistu",
  "valintakokeet": Array [
    Object {
      "id": undefined,
      "metadata": Object {
        "erityisjarjestelytMahdollisia": true,
        "liittyyEnnakkovalmistautumista": true,
        "ohjeetEnnakkovalmistautumiseen": Object {
          "fi": "<p>Ohjeet ennakkovalmistautumiseen - fi</p>",
          "sv": "<p>ohjeet ennakkovalmistautumiseen - sv</p>",
        },
        "ohjeetErityisjarjestelyihin": Object {
          "fi": "<p>Ohjeet erityisj채rjestelyihin - fi</p>",
          "sv": "<p>Ohjeet erityisj채rjestelyihin - sv</p>",
        },
        "tietoja": Object {
          "fi": "<p>Tietoa hakijalle - fi</p>",
          "sv": "<p>Tietoa hakijalle - sv</p>",
        },
        "vahimmaispisteet": 30.4,
      },
      "nimi": Object {
        "fi": "nimi - fi",
        "sv": "nimi - sv",
      },
      "tilaisuudet": Array [
        Object {
          "aika": Object {
            "alkaa": "2019-04-16T08:44",
            "paattyy": "2019-04-18T08:44",
          },
          "jarjestamispaikka": Object {
            "fi": "jarjestamispaikka - fi",
            "sv": "jarjestamispaikka - sv",
          },
          "lisatietoja": Object {
            "fi": "<p>fi lisatietoja</p>",
            "sv": "<p>sv lisatietoja</p>",
          },
          "osoite": Object {
            "osoite": Object {
              "fi": "fi osoite",
              "sv": "sv osoite",
            },
            "postinumeroKoodiUri": "posti_1#1",
          },
        },
      ],
      "tyyppiKoodiUri": "tyyppi_1#1",
    },
  ],
}
`;
